components/automate-gateway/api/iam/v2beta/rules.proto

```diff
@@ -15,56 +15,45 @@ import "components/automate-grpc/protoc-gen-policy/iam/annotations.proto";
+    option (chef.automate.api.policy).action = "update";
+    option (chef.automate.api.iam.policy).resource = "iam:projects:{project_id}";
+    option (chef.automate.api.iam.policy).action = "iam:projects:update";
   };
 
   rpc UpdateRule (UpdateRuleReq) returns (UpdateRuleResp) {
-    option (google.api.http).put = "/iam/v2beta/rules/{id}";
+    option (google.api.http).put = "/iam/v2beta/projects/{project_id}/rules/{id}";
     option (google.api.http).body = "*";
-    option (chef.automate.api.policy).resource = "auth:rules:{id}";
+    option (chef.automate.api.policy).resource = "auth:projects:{project_id}";
     option (chef.automate.api.policy).action = "update";
-    option (chef.automate.api.iam.policy).resource = "iam:rules:{id}";
-    option (chef.automate.api.iam.policy).action = "iam:rules:update";
+    option (chef.automate.api.iam.policy).resource = "iam:projects:{project_id}";
+    option (chef.automate.api.iam.policy).action = "iam:projects:update";
   };
 
   rpc GetRule (GetRuleReq) returns (GetRuleResp) {
-    option (google.api.http).get = "/iam/v2beta/rules/{id}";
-    option (chef.automate.api.policy).resource = "auth:rules:{id}";
+    option (google.api.http).get = "/iam/v2beta/projects/{project_id}/rules/{id}";
+    option (chef.automate.api.policy).resource = "auth:projects:{project_id}";
     option (chef.automate.api.policy).action = "get";
-    option (chef.automate.api.iam.policy).resource = "iam:rules:{id}";
-    option (chef.automate.api.iam.policy).action = "iam:rules:get";
-  };
-
-  // TODO (tc): Right now we don't have plans to use this in the UI
-  // and it doesn't return the staged rules like all the other query APIs.
-  // Should probably delete it or update it to return staged rules.
-  rpc ListRules (ListRulesReq) returns (ListRulesResp) {
-    option (google.api.http).get = "/iam/v2beta/rules";
-    option (chef.automate.api.policy).resource = "auth:rules";
-    option (chef.automate.api.policy).action = "read";
-    option (chef.automate.api.iam.policy).resource = "iam:rules";
-    option (chef.automate.api.iam.policy).action = "iam:rules:list";
+    option (chef.automate.api.iam.policy).resource = "iam:projects:{project_id}";
+    option (chef.automate.api.iam.policy).action = "iam:projects:get";
   };
 
   rpc ListRulesForProject (ListRulesForProjectReq) returns (ListRulesForProjectResp) {
     option (google.api.http).get = "/iam/v2beta/projects/{id}/rules";
-    option (chef.automate.api.policy).resource = "auth:projects:{id}:rules";
-    option (chef.automate.api.policy).action = "read";
-    option (chef.automate.api.iam.policy).resource = "iam:projects:{id}:rules";
-    option (chef.automate.api.iam.policy).action = "iam:rules:list";
+    option (chef.automate.api.policy).resource = "auth:projects:{id}";
+    option (chef.automate.api.policy).action = "get";
+    option (chef.automate.api.iam.policy).resource = "iam:projects:{id}";
```

Comment:
Should use `project_id` for consistency, no?

Reply:
We thought that this fits with the convention of using `id` to identify the relevant resource in any API. (in this case, it's the `id` of the project). In the APIs with two IDs, we chose to use `project_id` to differentiate from the rule `id`, since the rule is specific resource being operated on. We also went this way to minimize cascading backend changes. 

That being said, in writing this PR, @blakestier and I saw that it would make a lot of sense to bring all of these APIs under the project API (in which case we might just always use `id` for the project as the primary resource). I'll add a tech debt card to do so.